velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -502,11 +502,15 @@ std::pair<std::string, std::string> HiveDataSink::getWriterFileNames(
     targetFileName = computeBucketedFileName(
         connectorQueryCtx_->queryId(), bucketId.value());
   } else {
+    // targetFileName includes planNodeId and Uuid. As a result, different table
+    // writers run by the same task driver or the same table writer run in
+    // different task tries would have different targetFileNames.
     targetFileName = fmt::format(
-        "{}_{}_{}",
+        "{}_{}_{}_{}",
         connectorQueryCtx_->taskId(),
         connectorQueryCtx_->driverId(),
-        isCommitRequired() ? "0" : makeUuid());
+        connectorQueryCtx_->planNodeId(),
+        makeUuid());
   }
   const std::string writeFileName = isCommitRequired()
       ? fmt::format(".tmp.velox.{}_{}", targetFileName, makeUuid())
```

Comment:
Curious, what is the purpose of appending `uuid` in the final fileName ?

Reply:
Hi @shrinidhijoshi , including planNodeId and Uuid in the final fileName so that different table writers run by the same task driver or the same table writer run in different task attempts would have different final fileName. This is actually to have the approach consistent with the way that PrestoSpark Java composes the targetFileName, like https://github.com/prestodb/presto/blob/3fbbb677da4c9dc2386379d6a9d523fc2ae3c1c8/presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java#L356.

Reply:
Thanks for clarifying. My guess it is not needed on Sapphire Velox.
Because for the 2 use-cases you mentioned,
1.  different table writers run by the same task driver - This one would be distinguished by the `planFragmentId`
2.  same table writer run in different task attempts - This would be distinguished by the `attemptNumber` that is part of `taskId` field. This was a recent update https://github.com/prestodb/presto/pull/19649

wdyt ?

Reply:
Hi @shrinidhijoshi , you're absolutely right. As of now I haven't seen a case that needs randomness in targetFileName, for Sapphire Velox or for TaskCommit strategy. That made me start with the initial format w/o random. This time adding the uuid here only to keep consistent with Java, and probably avoid future problem.